### PR TITLE
Update RELEASE_NOTES.md

### PR DESCRIPTION
**Pull request by rfrost-hpe**
# Description

# Checklist

- [ ] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [ ] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [ ] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

 
[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -7,6 +7,19 @@ documentation improvements. This page lists some of the highlights.
 
 ### New Software Support
 
+* Resolved CASTS:
+    * `CAST-31486`: `cray-powerdns-manager` not adding compute nodes to external DNS
+    * `CAST-31712`: Unbound slamming DNS server with requests
+    * `CAST-31867`: Fixed Unbound not publishing to Prometheus
+    * `CAST-32362`: Unable to to monitor the progress of a boot of single node in the power on process when using BOS V2
```
Comment (mharding-hpe) [inline RELEASE_NOTES.md]: This is kind of misleading. Reading this, I would assume that BOS v2 had been changed in some way to allow this. However, I believe we actually just added some documentation to help customers collect the desired information. And that documentation probably went into CSM 1.4 as well, so it's not really part of CSM 1.5, per se.

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -7,6 +7,19 @@ documentation improvements. This page lists some of the highlights.
...
+    * `CAST-32978`: `prerequisites.sh` makes DNS not functional for some minutes
+    * `CAST-33450`: Worker rebuild workflow: `update-bss-no-wipe`
```
Comment (mharding-hpe) [inline RELEASE_NOTES.md]: This description does not do a good job of conveying what was fixed or changed, IMO.

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -7,6 +7,19 @@ documentation improvements. This page lists some of the highlights.
...
+    * `CAST-33608`: `spire` server slow to respond and crashing
+    * `CAST-33719`: Missing entries in `haproxy` on storage nodes
+    * `CAST-33922`: `PXE-E18: Server response timeout`
+    * `CAST-34613`: DNS has forgotten about all Compute/Application nodes
+    * `CAST-34633`: Add worker nodes to running system
```
Comment (mharding-hpe) [inline RELEASE_NOTES.md]: This description does not do a good job of conveying what was fixed or changed, IMO.

**Review by mharding-hpe — CHANGES_REQUESTED** [review]
Is this supposed to be a list of things fixed in CSM 1.5 that are not fixed in any previous CSM release? Or is it things that were fixed in CSM 1.5 but were still present in the product in CSM 1.4.0 (that is, things that may have also been fixed in 1.4.x patch releases)? I can see arguments for either one.

If this list is supposed to be stuff that is not fixed in any previous CSM releases (patch or otherwise), then a bunch of these do not qualify, as they have been fixed in previous CSM releases:

* `CAST-33450` Fixed in CSM 1.4.1
* `CAST-34633`: Fixed in CSM 1.3 and 1.4
* `CAST-31712`: Fixed in CSM 1.4.2
* `CAST-31867`: Fixed in CSM 1.4.0
* `CAST-32978`: Fixed in CSM 1.3.4 and 1.4.1
* `CAST-33608`: Fixed in CSM 1.4.3
* `CAST-34613`: Fixed in CSM 1.3.4 and 1.4.1
* `CAST-34692`: Fixed in CSm 1.4.4

File: RELEASE_NOTES.md [hunk 4]
```diff
@@ -7,6 +7,11 @@ documentation improvements. This page lists some of the highlights.
 
 ### New Software Support
 
+* Resolved CASTS:
+    * `CAST-31486`: `cray-powerdns-manager` not adding compute nodes to external DNS
+    * `CAST-33719`: Missing entries in `haproxy` on storage nodes
+    * `CAST-33922`: `PXE-E18: Server response timeout`
+
```
Comment (mharding-hpe) [inline RELEASE_NOTES.md]:
```suggestion
```

**Review by mharding-hpe — APPROVED** [review]

**Review by rustydb — APPROVED** [review]

**Pull request merged**